include/nbl/builtin/hlsl/spirv_intrinsics/core.hlsl

```diff
@@ -119,6 +180,44 @@ template<typename Signed>
 [[vk::ext_instruction( spv::OpBitFieldSExtract )]]
 Signed bitFieldSExtract( Signed val, uint32_t offsetBits, uint32_t numBits );
 
+}
+
+// should be in `nbl::hlsl::spirv` but would collide right now
+namespace experimental
+{
+using namespace nbl::hlsl::spirv;
+
+template<typename T, typename U>
+[[vk::ext_capability(spv::CapabilityPhysicalStorageBufferAddresses)]]
+[[vk::ext_instruction(spv::OpBitcast)]]
+T bitcast(U);
+
+template<typename T, uint32_t StorageClass>
+[[vk::ext_capability(spv::CapabilityPhysicalStorageBufferAddresses)]]
+[[vk::ext_instruction(spv::OpBitcast)]]
+uint64_t bitcast(pointer_t<StorageClass,T>);
+
+template<typename T, uint32_t StorageClass>
+[[vk::ext_capability(spv::CapabilityPhysicalStorageBufferAddresses)]]
+[[vk::ext_instruction(spv::OpBitcast)]]
+pointer_t<StorageClass,T> bitcast(uint64_t);
```

Comment:
in the case of a conversion fron/to `uint64_t` the `StorageClass` can only ever be the PhysicaStorage (BDA) there's literally no choice

Reply:
Done